lib/OnyxUtils.ts

```diff
@@ -783,71 +711,59 @@ function keyChanged(key, data, prevData, canUpdateSubscriber = () => true, notif
  * Sends the data obtained from the keys to the connection. It either:
  *     - sets state on the withOnyxInstances
  *     - triggers the callback function
- *
- * @private
- * @param {Object} mapping
- * @param {Object} [mapping.withOnyxInstance]
- * @param {String} [mapping.statePropertyName]
- * @param {Function} [mapping.callback]
- * @param {String} [mapping.selector]
- * @param {*|null} val
- * @param {String|undefined} matchedKey
- * @param {Boolean} isBatched
  */
-function sendDataToConnection(mapping, val, matchedKey, isBatched) {
+function sendDataToConnection<TKey extends OnyxKey>(
+    mapping: Mapping<TKey>,
+    val: OnyxCollection<OnyxValue<TKey>> | OnyxEntry<OnyxValue<TKey>>,
```

Comment:
```suggestion
    val: OnyxValue<TKey>,
```

In this example makes sense to use `OnyxValue`

Reply:
I have type errors after this change, any idea how to fix it?

Reply:
Other than this: https://github.com/Expensify/react-native-onyx/pull/523/commits/6807e085aa4f566208a91616b30a670c0f0bda86

Reply:
It's okay in my opinion (i do the same on `useOnyx`)